registry/client/repository.go

```diff
@@ -604,6 +607,25 @@ func (bs *blobs) Resume(ctx context.Context, id string) (distribution.BlobWriter
 	panic("not implemented")
 }
 
+func (bs *blobs) Mount(ctx context.Context, sourceRepo distribution.Repository, dgst digest.Digest) (distribution.Descriptor, error) {
+	u, err := bs.ub.BuildBlobUploadURL(bs.name, url.Values{"from": {sourceRepo.Name()}, "mount": {dgst.String()}})
+	if err != nil {
+		return distribution.Descriptor{}, err
+	}
+
+	resp, err := bs.client.Post(u, "", nil)
+	if err != nil {
+		return distribution.Descriptor{}, err
+	}
+	defer resp.Body.Close()
+
+	if SuccessStatus(resp.StatusCode) {
+		// TODO(brianbland): Check for possible Accepted response (legacy create)
+		return bs.Stat(ctx, dgst)
+	}
+	return distribution.Descriptor{}, handleErrorResponse(resp)
```

Comment:
Would expect `distribution.ErrBlobUnknown` or similar error when the blob could not be mounted because it did not exist for the repo.